### DISCUSSION TITLE
paste from system clipboard not working under linux (Ubuntu)

### DISCUSSION THREAD

**Discussion by umazanatastatura** [helix-editor/helix#5584] [General] [answered]
Hi,

I'm using helix 22.12 (96ff64a8) on
5.19.0-26-generic Ubuntu 22.10

and hx --health | head shows:
Clipboard provider: termcode
System clipboard provider: termcode

Terminal: GNOME Terminal Version 3.46.2 for GNOME 43
          Using VTE version 0.70.0 +BIDI +GNUTLS +ICU +SYSTEMD

Issue: Nothing happens when I paste text from system clipboard.
Observation: I've tried to use fish shell and pasting under it works, so I assume it's because of "system clipboard provider"..

**Comment by kirawi** [comment]
https://github.com/helix-editor/helix/wiki/Troubleshooting#on-linux
Next time, please use the bug template.

**Comment by the-mikedavis** [comment, marked as answer]
The termcode provider can't paste any values you didn't yank with `<space>y`/`<space>Y` within helix: https://github.com/helix-editor/helix/blob/0c51201ef21f567446746ae75950aa97b2bd6681/helix-view/src/clipboard.rs#L223-L247

You should install a clipboard provider program (see the Wiki link kirawi posted) for paste to work as you expect. For Wayland this would be `wl-clipboard`.
